docs/configuring-playbook-matrix-ldap-registration-proxy.md

```diff
@@ -0,0 +1,30 @@
+# Setting up matrix-ldap-registration-proxy (optional)
+
+The playbook can install and configure [matrix-ldap-registration-proxy](https://gitlab.com/activism.international/matrix_ldap_registration_proxy) for you.
+
+This proxy handles Matrix registration requests and forwards them to LDAP.
+
+**Please note:** This does support the full Matrix specification for registrations. It only provide a very coarse
+implementation of a basic password registration.
+
+## Quickstart
+
+Add the following configuration to your `inventory/host_vars/matrix.DOMAIN/vars.yml` file:
+
+```yaml
+matrix_ldap_registration_proxy_enabled: true
+```
+
+That is enough if you use the synapse external password provider via LDAP.
```

Comment:
```suggestion
This is enough **if** you use the [synapse external password provider via LDAP](docs/configuring-playbook-ldap-auth.md) (that is, you have `matrix_synapse_ext_password_provider_ldap_enabled: true` and other options in your configuration).
```

Still, as discussed [here](https://github.com/spantaleev/matrix-docker-ansible-deploy/pull/2038#discussion_r963057592) I wonder if this is a good idea.

Are these 2 things used together often? How does it help if you use both at the same time? If there is a benefit, we should probably document it as well. I bet lots of people will wonder "which one do I use? how is this one better? is it better still if I enable both at the same time?".

My understanding is that:

- the Synapse LDAP auth provider just handles login requests and tries to authenticate them against an LDAP server
- ldap-registration-proxy handles registration requests and creates users on the LDAP server (open registration?)

.. so they do complement each other and it's useful to have both at the same time (if you'd like both login and open registration functionality).

Reply:
I have the same understanding of the functionality. So I'd say there is no benefit to use them together, it is just possible. If you do, it could be an option (but not a good one if I think about it opinion) to reuse the credentials of Synapse LDAP auth provider. I think it would be better to have both roles specify which credentials to use and I'll add that to the docs and check in the `validate_config.yml ` if the values are set. An advanced user could then set these to 
```
matrix_ldap_registration_proxy_ldap_uri: "{{ matrix_synapse_ext_password_provider_ldap_uri }}"
matrix_ldap_registration_proxy_ldap_base_dn: "{{ matrix_synapse_ext_password_provider_ldap_base }}"
matrix_ldap_registration_proxy_ldap_user: "{{ matrix_synapse_ext_password_provider_ldap_bind_dn }}"
matrix_ldap_registration_proxy_ldap_password: "{{ matrix_synapse_ext_password_provider_ldap_bind_password }}"
```

Reply:
Isn't the benefit that one handles logins and the other handles new registrations (for an open registration server)?

Reply:
Exactly, that's the idea.

Reply:
But as far as I know they work independently from each other so I would not tie them together in `group_vars`